core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -955,6 +978,121 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext with Eventu
+      "transitive=true&transitive=false&transitive=invalidValue")
+    assert(sc.listJars().exists(_.contains("org.apache.hive_hive-storage-api-2.7.0.jar")))
+    assert(!sc.listJars().exists(_.contains("commons-lang_commons-lang-2.6.jar")))
+
+    sc.addJar("ivy://org.apache.hive:hive-storage-api:2.7.0?" +
+      "transitive=false&transitive=invalidValue&transitive=true")
+    assert(sc.listJars().exists(_.contains("org.apache.hive_hive-storage-api-2.7.0.jar")))
+    assert(sc.listJars().exists(_.contains("commons-lang_commons-lang-2.6.jar")))
+  }
+
+  test("SPARK-33084: Add jar support ivy url -- test param case sensitive") {
+    sc = new SparkContext(new SparkConf().setAppName("test").setMaster("local-cluster[3, 1, 1024]"))
+    sc.addJar("ivy://org.apache.hive:hive-storage-api:2.7.0?TRANSITIVE=true")
+    assert(sc.listJars().exists(_.contains("org.apache.hive_hive-storage-api-2.7.0.jar")))
+    assert(!sc.listJars().exists(_.contains("commons-lang_commons-lang-2.6.jar")))
+
+    sc.addJar("ivy://org.apache.hive:hive-storage-api:2.7.0?transitive=true")
+    assert(sc.listJars().exists(_.contains("org.apache.hive_hive-storage-api-2.7.0.jar")))
+    assert(sc.listJars().exists(_.contains("commons-lang_commons-lang-2.6.jar")))
+  }
+
+  test("SPARK-33084: Add jar support ivy url -- test transitive value case sensitive") {
+    sc = new SparkContext(new SparkConf().setAppName("test").setMaster("local-cluster[3, 1, 1024]"))
+    sc.addJar("ivy://org.apache.hive:hive-storage-api:2.7.0?transitive=TRUE")
+    assert(sc.listJars().exists(_.contains("org.apache.hive_hive-storage-api-2.7.0.jar")))
+    assert(!sc.listJars().exists(_.contains("commons-lang_commons-lang-2.6.jar")))
+
+    sc.addJar("ivy://org.apache.hive:hive-storage-api:2.7.0?transitive=true")
+    assert(sc.listJars().exists(_.contains("org.apache.hive_hive-storage-api-2.7.0.jar")))
+    assert(sc.listJars().exists(_.contains("commons-lang_commons-lang-2.6.jar")))
+  }
+
+
+  test("SPARK-33084: Add jar support ivy url -- test invalid ivy URI") {
+    sc = new SparkContext(new SparkConf().setAppName("test").setMaster("local-cluster[3, 1, 1024]"))
+    val e1 = intercept[IllegalArgumentException] {
+      sc.addJar("ivy://")
+    }.getMessage
+    assert(e1.contains("Expected authority at index 6: ivy://"))
+
+    val e2 = intercept[URISyntaxException] {
+      sc.addJar("ivy://org.apache.hive:hive-contrib")
+    }.getMessage
+    assert(e2.contains("Invalid url: Expected 'org:module:version'," +
+      " found org.apache.hive:hive-contrib"))
+
+    val e3 = intercept[URISyntaxException] {
+      sc.addJar("ivy://org.apache.hive:hive-contrib:2.3.7?foo=")
+    }.getMessage
+    assert(e3.contains("Invalid query string: foo="))
+
+    val e4 = intercept[URISyntaxException] {
+      sc.addJar("ivy://org.apache.hive:hive-contrib:2.3.7?bar=&baz=foo")
+    }.getMessage
+    assert(e4.contains("Invalid query string: bar=&baz=foo"))
+
+    val e5 = intercept[URISyntaxException] {
```

Comment:
For consistent error handling, `URISyntaxException`->`IllegalArgumentException` looks better? I meant it looks better to catch library exception in the Spark side, and then re-throw it as `IllegalArgumentException`.

Reply:
> IllegalArgumentException

Yea, catch and re-throw updated